### DISCUSSION TITLE
프로젝트 패키지화

### DISCUSSION THREAD

**Discussion by nomatterjun** [Favor-Gift-Reminder/Favor-iOS#82] [Ideas]
오늘 시간이 조금 남아서 YAPP이나 새싹 프로젝트들을 살펴봤는데
전에 말했던 패키지화가 많이 적용되어 있더라고..?
UI / Network / Core 이런 식으로 패키지를 나눠서 관리하는 방법 있었잖아..!

근데 내가 말했던 Tuist를 사용하는 방법까지는 아직 적용을 안한 것 같고 (아마 더 큰 프로젝트들에서 사용되는듯?)
우리같은 소규모 프로젝트들에서는
SPM을 사용해서 모듈화하는 방법이 많이 사용되는 것 같아

### 장점
1. 기능별로 패키지를 구분하기 때문에 프로젝트 파일들을 관리하기 유용해진다.
2. 변경점이 가해진 패키지/타겟만 빌드가 이루어지기 때문에 빌드 속도가 매우 빨라진다.

일단 이 정도가 있는 것 같아..!
당연히 CocoaPods / Carthage로도 된다고는 하지만 SPM이 제일 유리할 거고..

그래서 적용해보고 싶은데 시간이 살짝 붕 뜬 지금이 제일 적기인 것 같아서 작업해보고 싶은데 어떻게 생각해?

https://tech.kakao.com/2022/06/02/swift-package-manager/

아마 이 게시물을 중점으로 진행해볼 것 같아

**Comment by nomatterjun** [comment]
음.. 하다보니깐 작업이 끝나부렸어..
Package-Test 브랜치 확인 한 번 부탁해..!

**Reply by EnnzoKim** [reply]
좋은 것 같아! 
그러면 이제부터 추가되는 파일은 Package 디렉토리 안에 파일을 생성하면 되는거지?
FavorUIKit은 UIKit과 관련된 파일들 인 것 같은데,

FavorCoreKit은 어떤 범주로 파일을 모아둔걸까?

**Reply by nomatterjun** [reply]
CoreKit에서는 보통 `Foundation`의 `Extension`이나 OS와 밀접하게 닿아있는 기능들(`PhotosUI`, `CoreGraphics` 같은), `Logger`도 여기에 들어간대..!
`BaseViewController` 같은 `protocol`이 아닌 `Base`들도 다 여기 들어간다고 하는데 우리는 아직 그런 역할을 하는게 딱 `BaseViewController` 뿐이라 자세한건 회의때 이야기해보면 좋을 것 같아

**Reply by nomatterjun** [reply]
그러면 공식 피쳐로 추가해서 브랜치 파고 지금 브랜치는 지우도록 할게!

**Reply by EnnzoKim** [reply]
오케이~!
